app/models/manageiq/providers/azure_stack/cloud_manager/event_parser.rb

```diff
@@ -7,7 +7,7 @@ def self.event_to_hash(event, ems_id)
     {
       :event_type => event_type(event),
       :source     => 'AZURESTACK',
-      :ems_ref    => full_data[:id].downcase,
+      :ems_ref    => full_data[:id]&.downcase,
```

Comment:
This was hidden by the fact that as_null_object would return `double("event")` for all method calls